src/deepsparse/loggers/helpers.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Optional, Tuple
+
+from pydantic import BaseModel
+
+
+"""
+Helpers functions for logging
+"""
+
+__all__ = ["possibly_extract_value", "match"]
+
+
+def possibly_extract_value(
+    value: Any, identifier_reminder: Optional[str] = None
+) -> Any:
+    """
+    Given a string of ("."-separated) sub_identifiers, access the items
+    inside `value`.
+
+    :param value: A data structure that may potentially hold "nested" values
+        of interest.
+    :param sub_identifiers: A string of "."-separated keys that are used to
+        access "nested" value of interest inside`value`.
+    :return: Value of interest
+    """
+    if not identifier_reminder:
+        return value
+
+    value = dict(value) if isinstance(value, BaseModel) else value
+
+    for reminder in identifier_reminder.split("."):
+        # TODO: Add the support for slicing
+        # e.g. pipeline_inputs.images[0,0,0,0]
```

Comment:
@bfineran let's sync on implementation here. I understand the logic, but I have a hard time as of now thinking about elegant implementation.

Reply:
iterating through the string may just be easiest, otherwise some crazy regex pattern would work, prefer the former for readability  - some very rough code:

```python
# value: any object
# extraction_target: string starting with `.` or `[`
def extract_value(value, extraction_target):
   while extraction_target:
        current_target_type = extraction_target[0]
        extraction_target = extraction_target[1:]
        if current_target_type == ".":
             stopping_point = min(extraction_target.index("."), extraction_target.index("]"))  # stop at next property or index
             current_target = extraction_target[stopping_point]
             extraction_target = extraction_target[stopping_point:]
             value = getattr(value, current_target)
       elif current_target_type == "[":
             stopping_point = extraction_target.index("]")
             current_target = extraction_target[stopping_point]
             extraction_target = extraction_target[stopping_point + 1:]
             value = get_index(value, current_target)  # base case would just be value[current_target] but need to also handle int indices and potentially slicing
        else:
            break
    return value
```

Reply:
Okay, let's have it as a separate PR. This is a bit more involved asI expected.